packages/app/pages/queue.tsx

```diff
@@ -0,0 +1,234 @@
+import styled from "styled-components";
+import {
+  Role,
+  Question,
+  OpenQuestionStatus,
+  ClosedQuestionStatus,
+  QuestionType,
+  QuestionStatus,
+  UserCourse,
+} from "@template/common";
+import QuestionForm from "../components/Queue/QuestionForm";
+import QueueList from "../components/Queue/QueueList";
+import StudentPopupCard from "../components/Queue/StudentPopupCard";
+import {
+  useCallback,
+  useState,
+  useContext,
+  useEffect,
+  useMemo,
+  Fragment,
+} from "react";
+import useSWR from "swr";
+import { API } from "@template/api-client";
+import { ProfileContext } from "../contexts/ProfileContextProvider";
+
+// TODO: replace this with profile role from endpoint
+const ROLE: Role = Role.TA;
+
+const Container = styled.div`
+  margin: 32px 64px;
+  @media (max-width: 768px) {
+    margin: 32px 24px;
+  }
+`;
+
+interface QueueProps {}
+
+export default function Queue({}: QueueProps) {
+  const { profile } = useContext(ProfileContext);
+  const [course, setCourse] = useState<UserCourse>(null);
+  const [queueId, setQueueId] = useState<number>(null);
+  const [questions, setQuestions] = useState<Question[]>([]);
+
+  // Student queue state variables
+  const [isJoining, setIsJoining] = useState<boolean>(false);
+  const [questionDraftId, setQuestionDraftId] = useState<number>(null);
+
+  // TA queue state variables
+  const [openPopup, setOpenPopup] = useState<boolean>(false);
+  const [helpingQuestions, setHelpingQuestions] = useState<Question[]>([]);
+  const [groupQuestions, setGroupQuestions] = useState<Question[]>([]);
+  const [currentQuestion, setCurrentQuestion] = useState<Question>(null);
+
+  useEffect(() => {
+    if (profile) {
+      const selectedCourse: UserCourse = profile.courses[0];
+      setCourse(selectedCourse);
+      setQueueId(selectedCourse.course.id);
+    }
+  }, [profile]);
+
+  useEffect(() => {
+    if (queueId) {
+      getQuestions();
+    }
+  }, [queueId]);
+
+  /**
+   * Gets the questions for this course
+   */
+  const getQuestions = () => {
+    if (queueId) {
+      API.questions.index(queueId).then((q: Question[]) => {
```

Comment:
can use async/await for less code:
```
const getQuestions = async () => {
  if (queueId) {
    const q = await API.questions.index(queueId);
    if (q) {
    ...
```

also, is it possible to use useSWR for this?

Reply:
We were having trouble figuring out how to use SWR for this because we have to wait until the `/profile` request is fulfilled from the `ProfileContext` in order to send a request for the questions. I can open an issue for this if we can't find a quick solution?

Reply:
Referenced here: #19